hypothesis-python/src/hypothesis/internal/reflection.py

```diff
@@ -219,6 +220,35 @@ def ast_arguments_matches_signature(args, sig):
     return expected == [(p.name, p.kind) for p in sig.parameters.values()]
 
 
+class _NameUsageNodeVisitor(ast.NodeVisitor):
+    """Node visitor to check if a name is used inside an AST."""
+
+    def __init__(self, f, name):
+        # Remove indentation the function source might have to allow ast to parse it.
+        source = textwrap.dedent(inspect.getsource(f))
+        self.tree = ast.parse(source)
```

Comment:
`inspect.getsource` or `ast.parse` might fail (weird but true!), so we'll want to catch an exception around that and only run the check if we got the tree.

Reply:
Right. Fixed in #87424d2be839333d9b975f81a58452d5c79a335e.

The docs say that `ast.parse` might throw a `ValueError` if the null character shows up in the source, but I don't think that case can happen with strings that come from a successful call to `inspect.getsource`. I do not know if there are other cases in which `ast.parse` throws a `ValueError`, so I still capture that error in the `try-except` block.

Reply:
I'd just `except Exception:`, because we might also get a `SyntaxError` etc. (e.g. from Cython code)